sycl/test/on-device/basic_tests/stream/stream_copies_buffer_sync.cpp

```diff
@@ -0,0 +1,47 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
+// RUN: env SYCL_DEVICE_TYPE=HOST %t.out | FileCheck %s
+// RUN: %CPU_RUN_PLACEHOLDER %t.out %CPU_CHECK_PLACEHOLDER
+// RUN: %GPU_RUN_ON_LINUX_PLACEHOLDER %t.out %GPU_CHECK_ON_LINUX_PLACEHOLDER
+// RUN: %ACC_RUN_PLACEHOLDER %t.out %ACC_CHECK_PLACEHOLDER
+
+//==-- stream_copies_buffer_sync.cpp - SYCL stream flush buffer sync test --==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <CL/sycl.hpp>
+
+using namespace cl::sycl;
+
+void dev_func(stream out) { out << "dev_func print\n"; }
+
+int main() {
+  {
+    queue Queue;
+
```

Comment:
Should one add a test case with attempt to exceed the limits?
Is there such test-case already present?

Reply:
Added a test on maximum statement limit exceeding.